File: be-arise/src/main/java/vn/codezx/arise/dtos/grade/GradeDTO.java
package vn.codezx.arise.dtos.grade;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import vn.codezx.arise.dtos.base.BaseInfoDTO;
import vn.codezx.arise.dtos.classArise.ClassDTO;
import vn.codezx.arise.dtos.student.StudentDTO;
import vn.codezx.arise.entities.course.TestType;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@SuperBuilder
public class GradeDTO extends BaseInfoDTO {
  private int id;
  private StudentDTO student;
  private ClassDTO classArise;
  private TestType testType;
  private String comment;
  private Float sum;
  private Float score;
  private String classification;
  private List<SkillDTO> skills;
  private CriteriaDTO criteria;
}

File: fe-arise/src/dtos/grade/GradeDTO.ts
import { AuditInfoDTO } from "../base";
import { ClassDTO } from "../classes/ClassDTO";
import { StudentDTO } from "../student";
import { CriteriaDTO } from "./CriteriaDTO";
import { SkillDTO } from "./SkillDTO";
import { TestTypeDTO } from "./TestTypeDTO";

export type GradeDTO = {
  id: number;
  student: StudentDTO;
  classArise: ClassDTO;
  testType: TestTypeDTO;
  comment: string;
  sum: number;
  score: number;
  classification: string;
  skills: SkillDTO[];
  criteria: CriteriaDTO;
} & AuditInfoDTO;